update dune 2

Et7f3 opened the pull request:
I have some trouble with travis ci. I don't this I have regression since I haven't touched source file (apart from formatting).
Maybe revert some of my commit related to ci :frowning:
I have added generate_opam_files so flex is now pinable with opam :) and merged esy.json wih package.json because it was mostly duplicate.
I have choosen dune 2.0 so we have always a version that is usable: <2 = master and with this PR we have >=2 support.
I have found a jbuild related file ... maybe old trashfile.
Some commit are bloated by lockfiles but they could be removed with github filter :) and last commit is just formatting so it can be cancelled easily.

PS: the CI test with node 6 and 8... and do nothing on linux (but I am on linux and I confirm it build :) )

Et7f3 committed:
Update lockfiles of esy

Et7f3 committed:
dune upgrade

Et7f3 committed:
merge esy.json with package.json

Et7f3 committed:
remove refmterr of dependencies

Et7f3 committed:
Add range for OCaml and remove peerDependencies

Et7f3 committed:
dune lang 1.1 -> 2.0

Et7f3 committed:
generate_opam_files + opam lint

Et7f3 committed:
remove test from final package

Et7f3 committed:
remove esy-installer and add esy format

Et7f3 committed:
esy format

jordwalke commented:
Thanks! CI is really tricky to get right and travis was always flaky and not cross plat. Maybe you could just copy what we have in hello-reason to this repo, since it's thew new CI workflow that we encourage. It's mostly copy/paste and should work for all projects (only thing you might need to add is an `esy docs` command - maybe not even that). I'll set up an Azure pipeline for this project.

jordwalke commented:
I just pushed the hello-reason CI to master branch which will test all platforms. It assumes you have an `esy test` command as well as a "release" section in the package.json, but you can delete the release portion from the `hello-reason` config but you can delete that part of the pipelines.

Et7f3 commented:
Removed commit https://github.com/Et7f3/flex/commit/59c85cc84ab58b10f763f81767c392b313c51061, https://github.com/Et7f3/flex/commit/8035b5d00c4960f8e4ce101719e7f38e4f62f1da and https://github.com/Et7f3/flex/commit/15a43cef615d058ceaa6825eee3c61b1d0dfc72f and rebased onto master.

Et7f3 committed:
remove doc and release part

jordwalke commented on src/measure-test-fixed-encoding/dune:
```diff
@@ -1,4 +1,3 @@
 (executable
  (name MeasureTestFixedEncoding)
- (public_name MeasureTestFixedEncoding)
```
Why was this removed?

Et7f3 replied:
Because it is used only during the dev time. We can still exec it see https://github.com/jordwalke/flex/blob/master/package.json#L39. With this change dune need to build 138 instead of 150 targets. It save the 3Mb of this exe.

jordwalke commented:
All green!
